base/iterators.jl

```diff
@@ -1155,6 +1155,24 @@ julia> [(x,y) for x in 0:1 for y in 'a':'c']  # collects generators involving It
 """
 flatten(itr) = Flatten(itr)
 
+"""
+    flatten(f, iterators...)
+
+Create a lazy flattened mapping. Equivalent to `flatten(Iterators.map(f, iterators...))`.
```

Comment:
It's not entirely equivalent to `Iterators.map` because the function argument that accepts is untyped whereas the one this accepts is constrained to be a `Function`. How bad would it be to do something like
```julia
flatten(f, itrs...) = applicable(iterate, f) ? flatten((f, itrs...)) : flatten(map(f, itrs...))
```
? An approach like this would end up "preferring" iterating in the case that the user passes in some godforsaken object that is both iterable and callable.

Reply:
Flatten only accepts a single argument right now, so it would be unambiguous to define `flatten(f, itrs...) = flatten(map(f. itrs...))`, but I suppose I wanted nicer error messages when someone calls `flatten([1,2,3], [4,5,6])`.